src/test/java/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,166 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.bookstore.OnlineBookstoreApplication;
+import com.bookstore.dto.BookDto;
+import com.bookstore.dto.CreateBookRequestDto;
+import com.bookstore.entity.Category;
+import com.bookstore.service.BookService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(classes = OnlineBookstoreApplication.class)
+@Sql(scripts = "classpath:database/books/add-three-default-books.sql",
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+class BookControllerTest {
+
+    private static MockMvc mockMvc;
+
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private BookService bookService;
+
+    @BeforeEach
+    void setUp(@Autowired WebApplicationContext applicationContext) {
+        objectMapper = new ObjectMapper();
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get all books")
+    void getAllBooks_ReturnsAllBooks_ExpectedSuccess() throws Exception {
+        MvcResult result = mockMvc.perform(
+                        get("/books")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .param("page", "0")
+                                .param("size", "5")
+                )
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.content").isArray())
+                .andReturn();
+
+        assertNotNull(result.getResponse().getContentAsString());
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get book by ID")
+    void getBookById_ReturnsBookById_ExpectedSuccess() throws Exception {
```

Comment:
Please, check everywhere if you can improve names of your test methods
```suggestion
    void getBookById_WithValidId_Success() throws Exception {
```